src/application-instance.ts

```diff
@@ -11,8 +11,15 @@ import User from "./resources/user";
 export default class ApplicationInstance {
   public user: User | undefined;
   public transaction: Knex.Transaction | NoOpTransaction;
+  public baseUrl?: URL | undefined;
 
-  constructor(public app: Application) { }
+  constructor(public app: Application, baseUrl?: URL) {
+    if (!app.baseUrl && !baseUrl) {
+      throw Error('Application "baseUrl" is not defined!');
+    }
+
+    this.baseUrl = app.baseUrl || baseUrl;
+  }
```

Comment:
I'm not sure I understand the reason for having `baseUrl` available from either  `app.baseUrl` or the `baseUrl` parameter? Can you clarify for me?

Reply:
We can build baseUrl from `req.protocol` and `req.host` and pass it to app instance. But if you run the API behind some proxy with different domain you maybe want to hardcode it or pass it through env. Also, WebSocket server needs the hardcoded value too.